### PR TITLE
Pre-commit and poetry tasks

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

This PR adds the following:
* Poetry task to perform testing
* Poetry task for quality checking
  * This currently includes flake8 and pylint, but we should go through and figure out resolving errors
* Pre-commit hooks
  * Also includes flake8 and pylint, but for now this is commented out. 

I have pointed this PR to poetry_setup rather than master as that should be merged first.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] Code has been run through [`black`](https://github.com/psf/black) with the `-l 79` flag.
- [ ] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

_PR template was adopted from [appium](https://github.com/appium/appium/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,6 +31,8 @@ poethepoet = "^0.16.2"
 
 [tool.poe.tasks]
 setup = "pre-commit install"
+quality = {shell = "isort afidsvalidator & black afidsvalidator -l 79 & flake8 afidsvalidator & pylint afidsvalidator" }
```
Comment (tkkuehn) [inline pyproject.toml:34]: I think the `&`s here should be `&&`s.

**Review by tkkuehn — CHANGES_REQUESTED** [review]
Mostly looks good to me, one comment

File: pyproject.toml (hunk 1 above)
Comment (kaitj) [reply pyproject.toml]: Fixed in 62329ef

**Comment by kaitj** [comment]
Did we want to try to resolve potential warnings as part of this PR or separate (the linting is likely still to fail or output a long list for now).

**Comment by tkkuehn** [comment]
Lets fix the rest of them in a separate PR

**Comment by tkkuehn** [comment]
Oh, should we use poethepoet in the quality check actions now?

**Comment by kaitj** [comment]
They actually do two different things. The task actually performs the sorting and corrections but the action only performs checks for isort and black. 

We could create a separate task that mimics the checking for the action.

**Comment by tkkuehn** [comment]
Yeah I don't think we'll want to perform the corrections in the workflow, so I'm fine either way.

**Pull request merged**